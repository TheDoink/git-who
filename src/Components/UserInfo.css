#userInfo {
	font-size: 3vw;
	/*display:inline-block;*/
	text-align: center;
	width:100%;
}

#userInfo #userName {
	font-size:3em;
}

#userInfo #realName {
	padding-top:5px;
	font-size:2em;
}

#userInfo #location {
	font-size:2em;
}

/* The person icon */
#personIcon {
	color: #078800;
	font-size: 30vw;
}

#iconRow {
	text-align: center;
	width:100%;
	margin-top:10px;
}

#userIcon {
	border-width: 2vw;
	border-style: solid;
	border-color: #078800;


	border-radius:2vw;
	-moz-border-radius:2vw;
}