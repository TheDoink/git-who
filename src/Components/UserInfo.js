import React, { Component } from 'react';
import './UserInfo.css';

class UserInfo extends Component {

  getUserIconStyle = () => {

    // Define a box where the image will go 
    let ret = {
      'width':'22vw',
      'height':'22vw',
      'backgroundPosition':'center',
      'backgroundSize':'contain',
      'backgroundRepeat': 'no-repeat',
      'display':'inline-block',
      'position':'relative'
    };

    // If we have a user...
    if(this.props.gitUser.avatar_url) {
      
      // Display that user's image
      ret['backgroundImage'] = `url('${this.props.gitUser.avatar_url}')`;
    } else {

      // Otherwise hide it
      ret.display = "none";
    }

    return ret;
  };

  render() {
    
    return (
        <div>
          <div id="iconRow">

            {/*A user's avatar*/}
            <div id="userIcon" style={this.getUserIconStyle()}></div>

            {/*The placeholder icon*/}
            <div id="personIcon" className={this.props.gitUser.login ? 'hidden' : ''}>
              <i className="fa fa-user"></i>
            </div>

          </div>

          {/*Fun facts about the user, only displaying the ones they have*/}
          <div id="userInfo" className={this.props.gitUser.login ? '' : 'hidden'}>
            <div id="userName">{this.props.gitUser.login}</div>
            <div id="realName" className={this.props.gitUser.name ? '' : 'hidden'}>({this.props.gitUser.name})</div>
            <div id="location" className={this.props.gitUser.location ? '' : 'hidden'}>From - {this.props.gitUser.location}</div>
          </div>
        </div>
    );
  }
}

export default UserInfo;
