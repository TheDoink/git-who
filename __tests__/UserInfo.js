// var NameLink = require('../src/Components/NameLink');
import React from 'react';
import renderer from 'react-test-renderer';
import UserInfo from '../src/Components/UserInfo';

const mockUser = {
  "login": "hank",
  "id": 8534,
  "avatar_url": "https://avatars1.githubusercontent.com/u/8534?v=4",
  "gravatar_id": "",
  "url": "https://api.github.com/users/hank",
  "html_url": "https://github.com/hank",
  "followers_url": "https://api.github.com/users/hank/followers",
  "following_url": "https://api.github.com/users/hank/following{/other_user}",
  "gists_url": "https://api.github.com/users/hank/gists{/gist_id}",
  "starred_url": "https://api.github.com/users/hank/starred{/owner}{/repo}",
  "subscriptions_url": "https://api.github.com/users/hank/subscriptions",
  "organizations_url": "https://api.github.com/users/hank/orgs",
  "repos_url": "https://api.github.com/users/hank/repos",
  "events_url": "https://api.github.com/users/hank/events{/privacy}",
  "received_events_url": "https://api.github.com/users/hank/received_events",
  "type": "User",
  "site_admin": false
};

it('lines up with the snapshot', () => {

  // Testing visible gists & repos
  let testMe = renderer.create(<UserInfo gitUser={mockUser}></UserInfo>).toJSON();
  expect(testMe).toMatchSnapshot();
});
