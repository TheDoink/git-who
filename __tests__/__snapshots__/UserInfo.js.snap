// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lines up with the snapshot 1`] = `
<div>
  <div
    id="iconRow"
  >
    <div
      id="userIcon"
      style={
        Object {
          "backgroundImage": "url('https://avatars1.githubusercontent.com/u/8534?v=4')",
          "backgroundPosition": "center",
          "backgroundRepeat": "no-repeat",
          "backgroundSize": "contain",
          "display": "inline-block",
          "height": "22vw",
          "position": "relative",
          "width": "22vw",
        }
      }
    />
    <div
      className="hidden"
      id="personIcon"
    >
      <i
        className="fa fa-user"
      />
    </div>
  </div>
  <div
    className=""
    id="userInfo"
  >
    <div
      id="userName"
    >
      hank
    </div>
    <div
      className="hidden"
      id="realName"
    >
      (
      )
    </div>
    <div
      className="hidden"
      id="location"
    >
      From - 
    </div>
  </div>
</div>
`;
